app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarViewModel.kt

```diff
@@ -168,23 +169,37 @@ sealed class VaultUnlockedNavBarAction {
  * Models events for the bottom tab of the vault unlocked portion of the app.
  */
 sealed class VaultUnlockedNavBarEvent {
+
+    abstract val tab: VaultUnlockedNavBarTab
+
     /**
      * Navigate to the Generator screen.
      */
-    data object NavigateToGeneratorScreen : VaultUnlockedNavBarEvent()
+    data object NavigateToGeneratorScreen : VaultUnlockedNavBarEvent() {
+        override val tab: VaultUnlockedNavBarTab = VaultUnlockedNavBarTab.Generator
+    }
 
     /**
      * Navigate to the Send screen.
      */
-    data object NavigateToSendScreen : VaultUnlockedNavBarEvent()
+    data object NavigateToSendScreen : VaultUnlockedNavBarEvent() {
+        override val tab: VaultUnlockedNavBarTab = VaultUnlockedNavBarTab.Send
+    }
 
     /**
      * Navigate to the Vault screen.
      */
-    data object NavigateToVaultScreen : VaultUnlockedNavBarEvent()
+    data object NavigateToVaultScreen : VaultUnlockedNavBarEvent() {
+        override val tab: VaultUnlockedNavBarTab = VaultUnlockedNavBarTab.Vault(
+            labelRes = R.string.my_vault,
+            contentDescriptionRes = R.string.my_vault,
```

Comment:
So these particular values are never actually used, we just care about the type, right?